aten/src/ATen/native/quantized/cpu/qnnpack/src/qnnpack/pack_block_sparse.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+#include <cstdint>
+#include <vector>
+
+#include <qnnpack/math.h>
+
+typedef struct {
+  std::vector<uint32_t> col_indices;
+  std::vector<uint32_t> row_values;
+  std::vector<uint8_t> values;
+  uint32_t col_block_size;
+  uint32_t row_block_siz{1};
+  void print() {
+    std::cout << "row ptr\n";
+    for (const auto& t : row_values) {
+      std::cout << t << ", ";
+    }
+    std::cout << std::endl;
+    std::cout << "col indices\n";
+    for (const auto& t : col_indices) {
+      std::cout << t << ", ";
+    }
+    std::cout << std::endl;
+    std::cout << "Actual values\n";
+    for (const auto& t : values) {
+      std::cout << (uint32_t)t << ", ";
+    }
+    std::cout << std::endl;
+  }
+} BCSRMatrix;
+
+BCSRMatrix generateBlockCSRMatrix(
+    const uint8_t* a,
+    const size_t N,
+    const size_t K,
+    const uint32_t col_block_size,
+    const uint8_t* zero_points) {
+  BCSRMatrix bcsr_mat;
+  bcsr_mat.row_values.resize(N);
+  // K must be > 0
+  const uint32_t num_blocks = (K + col_block_size - 1) / col_block_size;
+  const uint64_t num_blocks_full =
+    (K % col_block_size) == 0 ? num_blocks : num_blocks - 1;
+  const uint32_t remainder_elements = K % col_block_size;
+
+  bcsr_mat.row_values.clear();
```

Comment:
bcsr_mat.row_values.resize() followed by clear should not be necessary when followed by reserve() and operating on a newly constructed vector.  If you just want to pre-allocate memory without modifying the vector size then a call to reserve should be enough.

Reply:
Yeah this is an oversight. Thanks for catching. Will fix it.